### PR TITLE
Marketplace: Use dynamic plugins endpoint

### PR DESCRIPTION
**Pull request by gavande1**
#### Proposed Changes

This PR introduces a way for switching between static and dynamic product catalogs for fetching plugin details. 

#### Testing Instructions

- Append `?flags=marketplace-fetch-dynamic-plugin-details` to enable the feature. 
- If required clear the indexeddb cache from the browser
- Visit the paid plugin details page - `plugins/woocommerce-bookings/:site`
- Open the network tab on the browser console and check if the plugin details page is fetching details from the newly created endpoint for getting details from the shadow database. 
- Check if plugin details are displayed correctly. 

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: client/data/marketplace/use-wpcom-plugins-query.ts [hunk 1]
```diff
@@ -32,7 +34,7 @@ const fetchWPCOMPlugins = ( type: Type, searchTerm?: string, tag?: string ) => {
 
 	return wpcom.req.get(
 		{
-			path: plugisApiBase,
+			path: pluginsApiBase,
```
Comment (cpapazoglou) [inline client/data/marketplace/use-wpcom-plugins-query.ts:37]: good catch.

**Review by cpapazoglou — CHANGES_REQUESTED** [review]
The changes work as advertised. It seems though that you have enabled `marketplace-jetpack-plugin-search` for all environments?

Unrelated to those changes, we need to work with the icon and the ratings

CleanShot 2023-01-11 at 12 02 28@2x

File: client/lib/plugins/utils.js [hunk 2]
```diff
@@ -240,7 +240,8 @@ export function normalizePluginData( plugin, pluginData ) {
 						item[ '2x' ] ||
 						item[ '1x' ] ||
 						item.svg ||
-						item.default;
+						item.default ||
+						item;
```
Comment (gavande1) [inline client/lib/plugins/utils.js:244]: The new backend returns the icon URL directly instead of wrapping it under the specific key.

**Review by cpapazoglou — APPROVED** [review]
LGTM!

**Pull request merged**